### PR TITLE
Refactor FXIOS-6594 [v116] Reorganize General & Account settings

### PR DESCRIPTION
**Pull request by lmarceau**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-6594)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/14763)

### Description
- This looks big but I copy pasted code mostly so we reorganize General & Account settings so each class have it's own file.
- Modified code in `AppSettingsTableViewController` to better organize instead of having one big function. Same functionality and settings order is kept.

### Pull requests checks where applicable
- [X] Fill in the three TODOs above (tickets number and description of your work)
- [X] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [X] Unit tests written and passing
- [ ] Documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Frontend/Settings/Main/Account/ConnectSetting.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Shared
+
+// Sync setting for connecting a Firefox Account. Shown when we don't have an account.
+class ConnectSetting: WithoutAccountSetting {
+    override var accessoryView: UIImageView? { return SettingDisclosureUtility.buildDisclosureIndicator(theme: theme) }
```
Comment (yoanarios) [inline Client/Frontend/Settings/Main/Account/ConnectSetting.swift:10]: nit: Break for long line
```suggestion
    override var accessoryView: UIImageView? { 
        return SettingDisclosureUtility.buildDisclosureIndicator(theme: theme) 
    }
```

File: Client/Frontend/Settings/Main/Account/ConnectSetting.swift [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    override var title: NSAttributedString? {
+        return NSAttributedString(string: .Settings.Sync.ButtonTitle, attributes: [NSAttributedString.Key.foregroundColor: theme.colors.textPrimary])
```
Comment (yoanarios) [inline Client/Frontend/Settings/Main/Account/ConnectSetting.swift:13]: nit: same comment as above break for long line

File: Client/Frontend/Settings/Main/DefaultBrowserSetting.swift [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+class DefaultBrowserSetting: Setting {
+    override var accessibilityIdentifier: String? { return "DefaultBrowserSettings" }
+
+    init(theme: Theme) {
+        super.init(title: NSAttributedString(string: String.DefaultBrowserMenuItem,
+                                             attributes: [NSAttributedString.Key.foregroundColor: theme.colors.actionPrimary]))
+    }
+
+    override func onClick(_ navigationController: UINavigationController?) {
+        TelemetryWrapper.gleanRecordEvent(category: .action, method: .open, object: .settingsMenuSetAsDefaultBrowser)
```
Comment (yoanarios) [inline Client/Frontend/Settings/Main/DefaultBrowserSetting.swift:17]: nit: same comment about long line

File: Client/Frontend/Settings/Main/General/NewTabPageSetting.swift [hunk 4]
```diff
@@ -0,0 +1,31 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+class NewTabPageSetting: Setting {
+    let profile: Profile
+
+    override var accessoryView: UIImageView? { return SettingDisclosureUtility.buildDisclosureIndicator(theme: theme) }
```
Comment (yoanarios) [inline Client/Frontend/Settings/Main/General/NewTabPageSetting.swift:10]: nit: same comment

File: Client/Frontend/Settings/Main/General/OpenWithSetting.swift [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Shared
+
+class OpenWithSetting: Setting {
+    let profile: Profile
+
+    override var accessoryView: UIImageView? { return SettingDisclosureUtility.buildDisclosureIndicator(theme: theme) }
```
Comment (yoanarios) [inline Client/Frontend/Settings/Main/General/OpenWithSetting.swift:11]: nit: Break for long line

File: Client/Frontend/Settings/Main/General/OpenWithSetting.swift [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+    override var accessibilityIdentifier: String? { return "OpenWith.Setting" }
+
+    override var status: NSAttributedString {
+        guard let provider = self.profile.prefs.stringForKey(PrefsKeys.KeyMailToOption) else {
+            return NSAttributedString(string: "")
+        }
+        if let path = Bundle.main.path(forResource: "MailSchemes", ofType: "plist"), let dictRoot = NSArray(contentsOfFile: path) {
```
Comment (yoanarios) [inline Client/Frontend/Settings/Main/General/OpenWithSetting.swift:19]: nit: Break for long line

File: Client/Frontend/Settings/Main/General/SearchSetting.swift [hunk 7]
```diff
@@ -0,0 +1,28 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+class SearchSetting: Setting {
+    let profile: Profile
+
+    override var accessoryView: UIImageView? { return SettingDisclosureUtility.buildDisclosureIndicator(theme: theme) }
```
Comment (yoanarios) [inline Client/Frontend/Settings/Main/General/SearchSetting.swift:10]: nit: same comment

File: Client/Frontend/Settings/Main/General/SiriPageSetting.swift [hunk 8]
```diff
@@ -0,0 +1,26 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+class SiriPageSetting: Setting {
+    let profile: Profile
+
+    override var accessoryView: UIImageView? { return SettingDisclosureUtility.buildDisclosureIndicator(theme: theme) }
```
Comment (yoanarios) [inline Client/Frontend/Settings/Main/General/SiriPageSetting.swift:10]: nit: same comment

File: Client/Frontend/Settings/Main/General/ThemeSetting.swift [hunk 9]
```diff
@@ -0,0 +1,33 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+class ThemeSetting: Setting {
+    let profile: Profile
+    override var accessoryView: UIImageView? { return SettingDisclosureUtility.buildDisclosureIndicator(theme: theme) }
```
Comment (yoanarios) [inline Client/Frontend/Settings/Main/General/ThemeSetting.swift:9]: nit: same comment

**Review by yoanarios — APPROVED** [review]
LGTM, some nit comments about breaking long lines. I know most of these comments are not related to your changes  but it will be nice to fix in this or future PR

**Comment by lmarceau** [comment]
Mh, failing UI tests. Looks like I might have broke something I'll need to investigate.

**Pull request merged**